ntoskrnl/se/token.c

```diff
@@ -4273,33 +4273,281 @@ NtDuplicateToken(
 
 /**
  * @brief
- * Changes the groups list of SIDs of a token.
+ * Private routine that iterates over the groups of an
```

Comment:
Is it meaningful to add such a full-blown documentation to a mere helper function, or instead to the main exposed API that uses it?

Reply:
Whilst the documentation of this private helper is full-blown, indeed, such documentation is only dedicated to that function. If one wants to know what `NtAdjustGroupsToken` does further, one has to check the code/doc/whatever of `SepAdjustGroups`. If your idea is to not have documentation at all for this helper and have everything moved to the syscall (even though there's a handful of private functions that have documentation for them already) then I'm rejecting your idea outright.

Reply:
np :)